lib/src/CoE/mailbox/response.cc

```diff
@@ -301,6 +303,217 @@ namespace kickcat::mailbox::response
+                sdo->fragments_left = requiered_fragments - fragment - 1;
+                if (sdo->fragments_left)
+                {
+                    sdo->incomplete = 1;
+                }
+
+                while ((header->len + sizeof(uint16_t) <= (data_.size() - sizeof(mailbox::Header))) and (pos < to_reply.size()))
+                {
+                    std::memcpy(data, to_reply.data() + pos, sizeof(uint16_t));
+                    pos += 1;
+                    data += sizeof(uint16_t);
+                    header->len += sizeof(uint16_t);
+                }
+                reply(std::move(raw_reply));
+
+                // Update counter handle
+                uint8_t counter = header_->count;
+                header_->count = mailbox::nextCounter(counter);
+            }
+        };
+
+        ListType list_type;
+        std::memcpy(&list_type, sdo_ + 1, sizeof(ListType));
+        switch (list_type)
+        {
+            case ListType::NUMBER:
+            {
+                uint16_t all_size      = static_cast<uint16_t>(dictionary.size());
+                uint16_t rxpdo_size    = 0;
+                uint16_t txpdo_size    = 0;
+                uint16_t backup_size   = 0;
+                uint16_t settings_size = 0;
+
+                for (auto const& object : dictionary)
+                {
+                    auto const& access = object.entries.at(0).access;
+                    if (access & CoE::Access::RxPDO)  { ++rxpdo_size;    }
+                    if (access & CoE::Access::TxPDO)  { ++txpdo_size;    }
+                    if (access & CoE::Access::BACKUP) { ++backup_size;   }
+                    if (access & CoE::Access::SETTING){ ++settings_size; }
+                }
+
+                uint8_t* data = reinterpret_cast<uint8_t*>(sdo_ + 1) + 2;
+                std::memcpy(data, &all_size,      sizeof(uint16_t)); data += sizeof(uint16_t);
+                std::memcpy(data, &rxpdo_size,    sizeof(uint16_t)); data += sizeof(uint16_t);
+                std::memcpy(data, &txpdo_size,    sizeof(uint16_t)); data += sizeof(uint16_t);
+                std::memcpy(data, &backup_size,   sizeof(uint16_t)); data += sizeof(uint16_t);
+                std::memcpy(data, &settings_size, sizeof(uint16_t)); data += sizeof(uint16_t);
+                header_->len += sizeof(ListType) + sizeof(uint16_t) * 5;
+                reply(std::move(data_));
+                break;
+            }
+            case ListType::ALL:     { fillList(list_type, CoE::Access::ALL);       break; }
+            case ListType::RxPDO:   { fillList(list_type, CoE::Access::RxPDO);     break; }
+            case ListType::TxPDO:   { fillList(list_type, CoE::Access::TxPDO);     break; }
+            case ListType::BACKUP:  { fillList(list_type, CoE::Access::BACKUP);    break; }
+            case ListType::SETTINGS:{ fillList(list_type, CoE::Access::SETTING);   break; }
+            default:
+            {
+                // TODO: unsupported access
```

Comment:
To be adressed in this PR ?

Reply:
I prefer to keep it: in a few months year, it will be easier to understand that this hole in checks was not filled on purpose ("wtf should we do here").